### PR TITLE
[VEGA-4144] fix(components): update row index for errors on row add

### PR DESCRIPTION
**Pull request by troynin**
## Problem statement

[РБ][FE] Индикация строк при добавлении соразмерно не сдвигается и теряется связь с динамическими изменениями
https://jira.vegaspace.tk/browse/VEGA-4144

## Solution details

Добавил изменения индекса строки для ошибок
stand: http://it-4144-lanit-vega-builder.code013.org

e2e: https://git.konakov.tv/vega2/vega-scenario-tests/-/merge_requests/81 

## Type

- [ ] Feature
- [x] Bug Fix
- [ ] Refactoring
- [x] Test
- [ ] Other

## Associated issues

_{Jira issues or PRs related to this one}_

## Quality control

- [ ] PR: Based on a correct branch
- [ ] PR: Head branch is rebased on the base branch
- [ ] PR: Assignee chosen and necessary labels are set
- [ ] PR: Current form is filled out (where it makes sense)
- [ ] PR: Diff checked for irrelevant changes
- [ ] PR: Commit messages follow [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) specs
- [ ] PR: Checked for spelling, grammar and typos
- [ ] JS: There's no errors/warnings in the browser dev console
- [ ] Layout: Tested with various content amounts
- [ ] Docs: All the important changes and features are described
- [ ] Tests: Configuration files are checked at test repositories
- [x] Tests: New unit tests developed
- [x] Tests: New E2E tests developed
- [x] Tests: Existing unit tests passed successfully

- [x] Tests: Existing E2E tests passed successfully

- [ ] Tests: Manually tested on personal instance

## Notable statements
- [ ] Affects configuration files
- [ ] Brings new packages or updates existing
- [ ] Opened follow-up tasks to resolve

### CODE REVIEW

File: src/components/ExcelTable/ExcelTable.test.tsx [hunk 1]
```diff
@@ -36,8 +38,33 @@ const tableState = {
 
 const mockSetRows = jest.fn();
 const mockSetColumns = jest.fn();
+const mockSetErrors = jest.fn();
 
-const renderComponent = (): RenderResult =>
+const errorsState: ColumnErrors = {
```
Comment (lipgen) [inline src/components/ExcelTable/ExcelTable.test.tsx:43]: думаю errorsState можно вынести в отдельный файл с моками

**Comment by pandamy619** [comment]
где ссылка на e2e ?

File: src/components/ExcelTable/ExcelTable.tsx [hunk 2]
```diff
@@ -304,6 +304,7 @@ export const ExcelTable: React.FC<ExcelTableProps> = ({
     { key }: RowContextBody,
   ): void => {
     const rowIndex = findRowIndexByKey(key, rows);
+    console.log('rowIndex', rowIndex);
```
Comment (lipgen) [inline src/components/ExcelTable/ExcelTable.tsx:307]: нужно подчистить

File: src/components/ExcelTable/utils/errors.ts [hunk 3]
```diff
@@ -34,6 +34,32 @@ function reduceRowIndexInKey(key: string, fromIndex: number): string {
   }
   return `row-${rowId - 1}`;
 }
+function increaseRowIndexInKey(key: string, fromIndex: number): string {
```
Comment (lipgen) [inline src/components/ExcelTable/utils/errors.ts:37]: покрой добавленные тобой функции юнит тестами

**Review by lipgen — APPROVED** [review]

**Pull request merged**